### PR TITLE
 Visibility options not updated when feedback path changes #5781

### LINKED ISSUE #5781: Visibility options not updated when feedback path changes
Bug introduced in #5671 because function in `.js` file is renamed, but the `onclick` attribute in `.tag` file not updated.

### PR DESCRIPTION
**Pull request by YongJieYongJie**
Fixes #5781

### CODE REVIEW

File: src/main/webapp/js/instructorFeedbackEdit.js [hunk 1]
```diff
@@ -523,7 +523,9 @@ function showNewQuestionFrame(type) {
     $('#empty_message').hide();
     scrollToElement($('#questionTableNew')[0], { duration: 1000 });
     $('#questionTableNew').find('.visibilityOptions').hide();
-    getVisibilityMessage($('#questionTableNew').find('.visibilityMessageButton'));
+
+    $newQuestionForm = $('#questionTableNew').closest('form');
+    updatePreviewTab($newQuestionForm);
```
Comment (wkurniawan07) [inline src/main/webapp/js/instructorFeedbackEdit.js]: May or may not be related, but isn't this procedure what `matchVisibilityOptionToFeedbackPath` does?

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java [hunk 2]
```diff
@@ -515,6 +522,66 @@ private void testChangeFeedbackRecipient() {
...
+        feedbackEditPage.clickQuestionEditForQuestion1();
+        feedbackEditPage.clickEditLabel(1);
+        feedbackEditPage.selectGiverToBe(FeedbackParticipantType.SELF, 1);
+        feedbackEditPage.selectRecipientToBe(FeedbackParticipantType.SELF, 1);
+
+        assertEnabledVisibilityOptionsIncludesOnly(Arrays.asList(
+                FeedbackParticipantType.RECEIVER, FeedbackParticipantType.OWN_TEAM_MEMBERS,
+                    FeedbackParticipantType.RECEIVER_TEAM_MEMBERS, FeedbackParticipantType.STUDENTS,
+                    FeedbackParticipantType.INSTRUCTORS),
+                1);
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java]: Would be more readable as:
```java
assertEnabledVisibilityOptionsIncludesOnly(
        Arrays.asList(FeedbackParticipantType.RECEIVER, FeedbackParticipantType.OWN_TEAM_MEMBERS,
                      FeedbackParticipantType.RECEIVER_TEAM_MEMBERS, FeedbackParticipantType.STUDENTS,
                      FeedbackParticipantType.INSTRUCTORS),
        1);
```
Similarly for all the assertions below.

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java [hunk 3]
```diff
@@ -515,6 +522,66 @@ private void testChangeFeedbackRecipient() {
...
+
+        feedbackEditPage.selectGiverToBe(FeedbackParticipantType.INSTRUCTORS, 1);
+        feedbackEditPage.selectRecipientToBe(FeedbackParticipantType.NONE, 1);
+
+        assertEnabledVisibilityOptionsIncludesOnly(Arrays.asList(
+                FeedbackParticipantType.STUDENTS, FeedbackParticipantType.INSTRUCTORS), 1);
+    }
+
+    private void assertEnabledVisibilityOptionsIncludesOnly(List<FeedbackParticipantType> expectedTypes,
+                                                           int questionNumber) {
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java]: Missing one space :p

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java [hunk 4]
```diff
@@ -515,6 +522,66 @@ private void testChangeFeedbackRecipient() {
...
+        Set<String> expectedEnabledOptions = new HashSet<String>();
+        for (FeedbackParticipantType expectedType : expectedTypes) {
+            expectedEnabledOptions.add(expectedType.toString());
+        }
+
+        Set<String> actualEnableOptions = new HashSet<String>();
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java]: I've nothing against using `Set`, but doesn't using `List` yield the same result?

**Comment by wkurniawan07** [comment]
Some changes requested.
Also I have confirmed that the test failed before the fix, passed after :+1:

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java (hunk 4 above)
Comment (YongJieYongJie) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java]: I actually chose `Set` because I didn't want the order of elements to matter.

**Comment by YongJieYongJie** [comment]
@wkurniawan07 Made the requested changes

File: src/main/webapp/js/instructorFeedbackEdit.js [hunk 5]
```diff
@@ -679,68 +681,9 @@ function copyOptions() {
...
-
-function feedbackGiverUpdateVisibilityOptions(elem) {
-    var $elem = $(elem);
-    if ($elem.val() === 'INSTRUCTORS' || $elem.val() === 'TEAMS') {
-        disableRow($elem, 2);
-        return;
-    }
-    enableRow($elem, 2);
+    $containingForm = $currGiver.closest('form');
+    updateEditTab($containingForm);
```
Comment (wkurniawan07) [inline src/main/webapp/js/instructorFeedbackEdit.js]: This can use the same `matchVisibilityOptionToFeedbackPath` function as well, right?

File: src/test/resources/pages/instructorFeedbackQuestionEditToTeamToTeam.html [hunk 6]
```diff
@@ -978,38 +978,42 @@ <h4 class="modal-title">
...
-                    <input class="visibilityCheckbox answerCheckbox1" type="checkbox" value="RECEIVER_TEAM_MEMBERS">
-                  </td>
-                  <td>
-                    <input class="visibilityCheckbox giverCheckbox1" type="checkbox" value="RECEIVER_TEAM_MEMBERS">
-                  </td>
-                  <td>
-                    <input class="visibilityCheckbox recipientCheckbox1" type="checkbox" value="RECEIVER_TEAM_MEMBERS">
-                  </td>
-                </tr>
+                <hide style="display: none;">
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorFeedbackQuestionEditToTeamToTeam.html]: Looks like I missed this before. Is this change intended?

Comment (wkurniawan07) [inline src/test/resources/pages/instructorFeedbackQuestionEditToTeamToTeam.html]: I just checked the PR #5671. Turns out this is a change that should have been caught in that PR (meaning we do have a test for it after all, although not really strong).
